### ISSUE TITLE
updating Scio from v0.8.1 to v.0.8.2 ends with RuntimeException

### ISSUE THREAD

**Issue by skysaeed** [spotify/scio#2778]
I recently tried to update Scio version from v0.8.1 to v.0.8.2 for a very simple streaming pipeline running on GCP dataflow but it continuously raises the following error as soon as the pipeline gets its first message from pubsub: 
```
java.lang.RuntimeException: org.apache.beam.vendor.guava.v26_0_jre.com.google.common.util.concurrent.UncheckedExecutionException: java.lang.ClassCastException: cannot assign instance of java.lang.invoke.SerializedLambda to field com.spotify.scio.util.Functions$$anon$3.g of type scala.Function1 in instance of com.spotify.scio.util.Functions$$anon$3
```

**Comment by regadas** [comment]
Hi @skysaeed thanks for the report.

Can you share the full stack trace? Also which scala version are you using?

**Comment by psilospore** [comment]
Ahhh got this too.
Here's my stacktrace
```
org.apache.beam.vendor.guava.v26_0_jre.com.google.common.util.concurrent.UncheckedExecutionException: java.lang.ClassCastException: cannot assign instance of java.lang.invoke.SerializedLambda to field com.spotify.scio.util.Functions$$anon$5.g of type scala.Function1 in instance of com.spotify.scio.util.Functions$$anon$5 org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory$1.typedApply(IntrinsicMapTaskExecutorFactory.java:194) org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory$1.typedApply(IntrinsicMapTaskExecutorFactory.java:165) org.apache.beam.runners.dataflow.worker.graph.Networks$TypeSafeNodeFunction.apply(Networks.java:63) org.apache.beam.runners.dataflow.worker.graph.Networks$TypeSafeNodeFunction.apply(Networks.java:50) org.apache.beam.runners.dataflow.worker.graph.Networks.replaceDirectedNetworkNodes(Networks.java:87) org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory.create(IntrinsicMapTaskExecutorFactory.java:125) org.apache.beam.runners.dataflow.worker.BatchDataflowWorker.doWork(BatchDataflowWorker.java:352) org.apache.beam.runners.dataflow.worker.BatchDataflowWorker.getAndPerformWork(BatchDataflowWorker.java:305) org.apache.beam.runners.dataflow.worker.DataflowBatchWorkerHarness$WorkerThread.doWork(DataflowBatchWorkerHarness.java:140) org.apache.beam.runners.dataflow.worker.DataflowBatchWorkerHarness$WorkerThread.call(DataflowBatchWorkerHarness.java:120) org.apache.beam.runners.dataflow.worker.DataflowBatchWorkerHarness$WorkerThread.call(DataflowBatchWorkerHarness.java:107) java.util.concurrent.FutureTask.run(FutureTask.java:266) java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) java.lang.Thread.run(Thread.java:748) Caused by: org.apache.beam.vendor.guava.v26_0_jre.com.google.common.util.concurrent.UncheckedExecutionException: java.lang.ClassCastException: cannot assign instance of java.lang.invoke.SerializedLambda to field com.spotify.scio.util.Functions$$anon$5.g of type scala.Function1 in instance of com.spotify.scio.util.Functions$$anon$5 org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2050) org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache.get(LocalCache.java:3952) org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache$LocalManualCache.get(LocalCache.java:4871) org.apache.beam.runners.dataflow.worker.UserParDoFnFactory.create(UserParDoFnFactory.java:91) org.apache.beam.runners.dataflow.worker.DefaultParDoFnFactory.create(DefaultParDoFnFactory.java:75) org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory.createParDoOperation(IntrinsicMapTaskExecutorFactory.java:264) org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory.access$000(IntrinsicMapTaskExecutorFactory.java:86) org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory$1.typedApply(IntrinsicMapTaskExecutorFactory.java:183) ... 14 more Caused by: java.lang.ClassCastException: cannot assign instance of java.lang.invoke.SerializedLambda to field com.spotify.scio.util.Functions$$anon$5.g of type scala.Function1 in instance of com.spotify.scio.util.Functions$$anon$5 java.io.ObjectStreamClass$FieldReflector.setObjFieldValues(ObjectStreamClass.java:2233) java.io.ObjectStreamClass.setObjFieldValues(ObjectStreamClass.java:1405) java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2288) java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2206) java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2064) java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1568) java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2282) java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2206) java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2064) java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1568) java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2282) java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2206) java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2064) java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1568) java.io.ObjectInputStream.readArray(ObjectInputStream.java:1970) java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1562) java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2282) java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2206) java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2064) java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1568) java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2282) java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2206) java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2064) java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1568) java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2282) java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2206) java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2064) java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1568) java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2282) java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2206) java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2064) java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1568) java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2282) java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2206) java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2064) java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1568) java.io.ObjectInputStream.readObject(ObjectInputStream.java:428) org.apache.beam.sdk.util.SerializableUtils.deserializeFromByteArray(SerializableUtils.java:71) org.apache.beam.runners.dataflow.worker.UserParDoFnFactory$UserDoFnExtractor.getDoFnInfo(UserParDoFnFactory.java:62) org.apache.beam.runners.dataflow.worker.UserParDoFnFactory.lambda$create$0(UserParDoFnFactory.java:93) org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache$LocalManualCache$1.load(LocalCache.java:4876) org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3528) org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2277) org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2154) org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2044)
```

**Comment by skysaeed** [comment]
> Hi @skysaeed thanks for the report.
> 
> Can you share the full stack trace? Also which scala version are you using?

Hi, I am using Scala 2.12.10. 
Here is the stack trace:
`
java.lang.RuntimeException: org.apache.beam.vendor.guava.v26_0_jre.com.google.common.util.concurrent.UncheckedExecutionException: java.lang.ClassCastException: cannot assign instance of java.lang.invoke.SerializedLambda to field com.spotify.scio.util.Functions$$anon$3.g of type scala.Function1 in instance of com.spotify.scio.util.Functions$$anon$3
        org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory$1.typedApply(IntrinsicMapTaskExecutorFactory.java:194)
        org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory$1.typedApply(IntrinsicMapTaskExecutorFactory.java:165)
        org.apache.beam.runners.dataflow.worker.graph.Networks$TypeSafeNodeFunction.apply(Networks.java:63)
        org.apache.beam.runners.dataflow.worker.graph.Networks$TypeSafeNodeFunction.apply(Networks.java:50)
        org.apache.beam.runners.dataflow.worker.graph.Networks.replaceDirectedNetworkNodes(Networks.java:87)
        org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory.create(IntrinsicMapTaskExecutorFactory.java:125)
        org.apache.beam.runners.dataflow.worker.StreamingDataflowWorker.process(StreamingDataflowWorker.java:1266)
        org.apache.beam.runners.dataflow.worker.StreamingDataflowWorker.access$1100(StreamingDataflowWorker.java:152)
        org.apache.beam.runners.dataflow.worker.StreamingDataflowWorker$7.run(StreamingDataflowWorker.java:1073)
        java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.beam.vendor.guava.v26_0_jre.com.google.common.util.concurrent.UncheckedExecutionException: java.lang.ClassCastException: cannot assign instance of java.lang.invoke.SerializedLambda to field com.spotify.scio.util.Functions$$anon$3.g of type scala.Function1 in instance of com.spotify.scio.util.Functions$$anon$3
        org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2050)
        org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache.get(LocalCache.java:3952)
        org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache$LocalManualCache.get(LocalCache.java:4871)
        org.apache.beam.runners.dataflow.worker.UserParDoFnFactory.create(UserParDoFnFactory.java:91)
        org.apache.beam.runners.dataflow.worker.DefaultParDoFnFactory.create(DefaultParDoFnFactory.java:75)
        org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory.createParDoOperation(IntrinsicMapTaskExecutorFactory.java:264)
        org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory.access$000(IntrinsicMapTaskExecutorFactory.java:86)
        org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory$1.typedApply(IntrinsicMapTaskExecutorFactory.java:183)
        org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory$1.typedApply(IntrinsicMapTaskExecutorFactory.java:165)
        org.apache.beam.runners.dataflow.worker.graph.Networks$TypeSafeNodeFunction.apply(Networks.java:63)
        org.apache.beam.runners.dataflow.worker.graph.Networks$TypeSafeNodeFunction.apply(Networks.java:50)
        org.apache.beam.runners.dataflow.worker.graph.Networks.replaceDirectedNetworkNodes(Networks.java:87)
        org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory.create(IntrinsicMapTaskExecutorFactory.java:125)
        org.apache.beam.runners.dataflow.worker.StreamingDataflowWorker.process(StreamingDataflowWorker.java:1266)
        org.apache.beam.runners.dataflow.worker.StreamingDataflowWorker.access$1100(StreamingDataflowWorker.java:152)
        org.apache.beam.runners.dataflow.worker.StreamingDataflowWorker$7.run(StreamingDataflowWorker.java:1073)
        java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.ClassCastException: cannot assign instance of java.lang.invoke.SerializedLambda to field com.spotify.scio.util.Functions$$anon$3.g of type scala.Function1 in instance of com.spotify.scio.util.Functions$$anon$3
        java.io.ObjectStreamClass$FieldReflector.setObjFieldValues(ObjectStreamClass.java:2233)
        java.io.ObjectStreamClass.setObjFieldValues(ObjectStreamClass.java:1405)
        java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2288)
        java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2206)
        java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2064)
        java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1568)
        java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2282)
        java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2206)
        java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2064)
        java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1568)
        java.io.ObjectInputStream.readObject(ObjectInputStream.java:428)
        org.apache.beam.sdk.util.SerializableUtils.deserializeFromByteArray(SerializableUtils.java:71)
        org.apache.beam.runners.dataflow.worker.UserParDoFnFactory$UserDoFnExtractor.getDoFnInfo(UserParDoFnFactory.java:62)
        org.apache.beam.runners.dataflow.worker.UserParDoFnFactory.lambda$create$0(UserParDoFnFactory.java:93)
        org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache$LocalManualCache$1.load(LocalCache.java:4876)
        org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3528)
        org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2277)
        org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2154)
        org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2044)
        org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache.get(LocalCache.java:3952)
        org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache$LocalManualCache.get(LocalCache.java:4871)
        org.apache.beam.runners.dataflow.worker.UserParDoFnFactory.create(UserParDoFnFactory.java:91)
        org.apache.beam.runners.dataflow.worker.DefaultParDoFnFactory.create(DefaultParDoFnFactory.java:75)
        org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory.createParDoOperation(IntrinsicMapTaskExecutorFactory.java:264)
        org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory.access$000(IntrinsicMapTaskExecutorFactory.java:86)
        org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory$1.typedApply(IntrinsicMapTaskExecutorFactory.java:183)
        org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory$1.typedApply(IntrinsicMapTaskExecutorFactory.java:165)
        org.apache.beam.runners.dataflow.worker.graph.Networks$TypeSafeNodeFunction.apply(Networks.java:63)
        org.apache.beam.runners.dataflow.worker.graph.Networks$TypeSafeNodeFunction.apply(Networks.java:50)
        org.apache.beam.runners.dataflow.worker.graph.Networks.replaceDirectedNetworkNodes(Networks.java:87)
        org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory.create(IntrinsicMapTaskExecutorFactory.java:125)
        org.apache.beam.runners.dataflow.worker.StreamingDataflowWorker.process(StreamingDataflowWorker.java:1266)
        org.apache.beam.runners.dataflow.worker.StreamingDataflowWorker.access$1100(StreamingDataflowWorker.java:152)
        org.apache.beam.runners.dataflow.worker.StreamingDataflowWorker$7.run(StreamingDataflowWorker.java:1073)
        java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        java.lang.Thread.run(Thread.java:748)`

**Comment by regadas** [comment]
Hi @skysaeed and @psilospore 

This stack traces are not very helpful unfortunately. It seems that this stack trace is a symptom of something else.

Can you provide an example of your pipeline?

**Comment by skysaeed** [comment]
Here is a code sample: 

`

	val topicPath: String = config.getString(ConfigKeys.topicPath)
	val topics: util.List[String] = config.getStringList(ConfigKeys.topics)

	def main(cargs: Array[String]): Unit = {

		val (sc, args) = ContextAndArgs(cargs)

		topics.forEach( topicName => {

		  val topic = s"$topicPath/$topicName"
		  val messages = sc.withName(topicName).pubsubTopic[String](topic).withFixedWindows(windowSize)
		  val jsonMessages = messages.flatMap({
		    element =>
		      parser.parse(element) match {
		        case Left(parsingFailure) =>
		          logger.error("parsing: {} failed.",element,parsingFailure)
		          None
		        case Right(json) =>
		          val configId = json.hcursor.downField("id").as[String].getOrElse("no-id")
		          Some(json.noSpaces, configId)
		      }
		  })
		}
	}
`

**Comment by jto** [comment]
Hum so `PubsubIO` has not changed between `0.8.1` and `0.8.2` so I think it may be a issue in beam. Still investigating.

**Comment by jto** [comment]
@skysaeed @psilospore I tried creating a new streaming job using `0.8.2` and I could not reproduce the issue. So I think it only happen if you're updating (`--update`) an already running job. 

```scala
 val (sc, args) = ContextAndArgs(cmdlineArgs)
    val opt = sc.options

    opt.setRunner(classOf[DataflowRunner])
    opt.setTempLocation("gs://<some bucket>/temp")

    val topic = "<a topic>"

    val stream =
      sc.withName("julient-test")
        .pubsubTopic[String](topic)
        .withFixedWindows(org.joda.time.Duration.standardSeconds(5))
        .debug()
        .map(x => println(x))

    sc.run.waitUntilDone(5 minutes)
```

Could you try renaming the `pubsubTopic` transform before updating ?

So for example

```scala
val topic = s"$topicPath/$topicName"
val messages = sc.withName(topicName).pubsubTopic[String](topic)
```

would become:

```scala
val topic = s"$topicPath/$topicName"
val messages = sc.withName(s"${topicName}-2").pubsubTopic[String](topic)
```

**Comment by psilospore** [comment]
@jto I wasn't using pubsub. I was reading from a bigquery table and then did some basic operations but that has changed since I got the bug. Not sure if I can recreate it since it's changed but will try to post an example.

**Comment by jto** [comment]
Ah I see. Thanks for the info. I was suspecting the problem to be in `withFixedWindows` so it seems to confirm my guess.

**Comment by skysaeed** [comment]
@jto 
It was a fresh deploy, but not update.
I investigated more  and found out that problem happens on the "map" or "flatMap" of Scollection. 
this code has no error : 
```
topics.forEach( topicName => {
      val topic = s"$topicPath/$topicName"
      val messages = sc.withName(topicName).pubsubTopic[String](topic).withFixedWindows(windowSize)
    })
```
but this one produces the error I have already reported :

```
topics.forEach( topicName => {
      val topic = s"$topicPath/$topicName"
      val messages = sc.withName(topicName).pubsubTopic[String](topic).withFixedWindows(windowSize)
      val jsons = messages.map(a => a)
    })
```

**Comment by skysaeed** [comment]
@jto @regadas 
I have to add a comment that this error is being generated on dataflow once the first message arrives from Pub/Sub

**Comment by jto** [comment]
@skysaeed Thanks for the code example.

Sadly I can't reproduce the issue... I'm leaning toward a bug in Beam because I don't see how the changes between Scio `0.8.1` and `0.8.2` could explain the bug.

Could you tell me how you are running the job  and which version of the JVM are you using ?

**Comment by skysaeed** [comment]
@jto Thanks,
I am using `jdk1.8.0_211`
Here is the command to run the job:
```
./gradlew :app-name:run --args=....
```
and the arguments are: 
```
--project=...
--runner=DataflowRunner
--numWorkers=1
--stagingLocation=gs://...
--tempLocation=gs://...
--region=...
--subnetwork=https://www.googleapis.com/...
```

For the minimal code in order to reproduce the error there are only three direct dependencies :

- org.scala-lang:scala-library:2.12.10

- com.spotify:scio-core_2.12:0.8.2

- org.apache.beam:beam-runners-google-cloud-dataflow-java:2.19.0


PS: Error is being generated only once the first message arrives on Pub/Sub

**Comment by jto** [comment]
@skysaeed it seems pretty similar to my test.

Could you try to force the version of `org.apache.beam:beam-runners-google-cloud-dataflow-java` to `2.18.0` and run the same job ? 

This is obviously not a fix but if the job run without throwing the exception, it would mean the issue is in the runner.

**Comment by skysaeed** [comment]
@jto , I tried to use runner 2.18.0, but it fails 
```
Exception in thread "main" java.lang.AbstractMethodError: org.apache.beam.runners.dataflow.PrimitiveParDoSingleFactory$PayloadTranslator$1.translateDoFn(Lorg/apache/beam/runners/core/construction/SdkComponents;)Lorg/apache/beam/model/pipeline/v1/RunnerApi$FunctionSpec;
	at org.apache.beam.runners.core.construction.ParDoTranslation.payloadForParDoLike(ParDoTranslation.java:738)
	at org.apache.beam.runners.dataflow.PrimitiveParDoSingleFactory$PayloadTranslator.payloadForParDoSingle(PrimitiveParDoSingleFactory.java:193)
	at org.apache.beam.runners.dataflow.PrimitiveParDoSingleFactory$PayloadTranslator.translate(PrimitiveParDoSingleFactory.java:154)
	at org.apache.beam.runners.core.construction.PTransformTranslation$KnownTransformPayloadTranslator.translate(PTransformTranslation.java:417)
	at org.apache.beam.runners.core.construction.PTransformTranslation.toProto(PTransformTranslation.java:225)
	at org.apache.beam.runners.core.construction.SdkComponents.registerPTransform(SdkComponents.java:157)
	at org.apache.beam.runners.core.construction.PipelineTranslation$1.visitPrimitiveTransform(PipelineTranslation.java:87)
	at org.apache.beam.sdk.runners.TransformHierarchy$Node.visit(TransformHierarchy.java:665)
	at org.apache.beam.sdk.runners.TransformHierarchy$Node.visit(TransformHierarchy.java:657)
	at org.apache.beam.sdk.runners.TransformHierarchy$Node.visit(TransformHierarchy.java:657)
	at org.apache.beam.sdk.runners.TransformHierarchy$Node.visit(TransformHierarchy.java:657)
	at org.apache.beam.sdk.runners.TransformHierarchy$Node.access$600(TransformHierarchy.java:317)
	at org.apache.beam.sdk.runners.TransformHierarchy.visit(TransformHierarchy.java:251)
	at org.apache.beam.sdk.Pipeline.traverseTopologically(Pipeline.java:460)
	at org.apache.beam.runners.core.construction.PipelineTranslation.toProto(PipelineTranslation.java:59)
	at org.apache.beam.runners.dataflow.DataflowPipelineTranslator.translate(DataflowPipelineTranslator.java:187)
	at org.apache.beam.runners.dataflow.DataflowRunner.run(DataflowRunner.java:797)
	at org.apache.beam.runners.dataflow.DataflowRunner.run(DataflowRunner.java:188)
	at org.apache.beam.sdk.Pipeline.run(Pipeline.java:315)
	at org.apache.beam.sdk.Pipeline.run(Pipeline.java:301)
	at com.spotify.scio.ScioContext.execute(ScioContext.scala:606)
	at com.spotify.scio.ScioContext$$anonfun$run$1.apply(ScioContext.scala:594)
	at com.spotify.scio.ScioContext$$anonfun$run$1.apply(ScioContext.scala:582)
	at com.spotify.scio.ScioContext.requireNotClosed(ScioContext.scala:702)
	at com.spotify.scio.ScioContext.run(ScioContext.scala:582)

```

and if I exclude beam libraries from Scio to get them from beam dataflow runner dependencies like :
```
    implementation (scio.scioCore){
        exclude group: 'org.apache.beam'
    }
```
then the error will be on Scio :
```
Exception in thread "main" java.lang.IllegalArgumentException: requirement failed: Mismatched version for DataflowRunner, expected: 2.19.0, actual: 2.18.0
	at scala.Predef$.require(Predef.scala:281)
	at com.spotify.scio.VersionUtil$.checkRunnerVersion(VersionUtil.scala:127)
	at com.spotify.scio.ScioContext.<init>(ScioContext.scala:456)
	at com.spotify.scio.ContextAndArgs$$anonfun$withParser$1.apply(ScioContext.scala:240)
	at com.spotify.scio.ContextAndArgs$$anonfun$withParser$1.apply(ScioContext.scala:230)
	at com.spotify.scio.ContextAndArgs$.apply(ScioContext.scala:245)
```

**Comment by jto** [comment]
Hey there. 

Bit of a heads up. We still don't know what's causing this issue BUT we do have a batch job internally that throws the exact same exception.
The problem is not specific to or pubsub.

**Comment by jgogstad** [comment]
Has this been observed in 0.9.0?

**Comment by stormy-ua** [comment]
FWIW, bbserved the same issue today. 

Environment: Scio 0.9.0, Scala 2.12.11, Beam SDK 2.20.0.

Error: 
```bash
SEVERE: 2020-05-15T16:55:04.641Z: java.lang.RuntimeException: org.apache.beam.vendor.guava.v26_0_jre.com.google.common.util.concurrent.UncheckedExecutionException: java.lang.ClassCastException: cannot assign instance of java.lang.invoke.SerializedLambda to field com.spotify.scio.util.Functions$$anon$5.g of type scala.Function1 in instance of com.spotify.scio.util.Functions$$anon$5
	at org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory$1.typedApply(IntrinsicMapTaskExecutorFactory.java:194)
	at org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory$1.typedApply(IntrinsicMapTaskExecutorFactory.java:165)
	at org.apache.beam.runners.dataflow.worker.graph.Networks$TypeSafeNodeFunction.apply(Networks.java:63)
	at org.apache.beam.runners.dataflow.worker.graph.Networks$TypeSafeNodeFunction.apply(Networks.java:50)
	at org.apache.beam.runners.dataflow.worker.graph.Networks.replaceDirectedNetworkNodes(Networks.java:87)
	at org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory.create(IntrinsicMapTaskExecutorFactory.java:125)
	at org.apache.beam.runners.dataflow.worker.BatchDataflowWorker.doWork(BatchDataflowWorker.java:353)
	at org.apache.beam.runners.dataflow.worker.BatchDataflowWorker.getAndPerformWork(BatchDataflowWorker.java:306)
	at org.apache.beam.runners.dataflow.worker.DataflowBatchWorkerHarness$WorkerThread.doWork(DataflowBatchWorkerHarness.java:140)
	at org.apache.beam.runners.dataflow.worker.DataflowBatchWorkerHarness$WorkerThread.call(DataflowBatchWorkerHarness.java:120)
	at org.apache.beam.runners.dataflow.worker.DataflowBatchWorkerHarness$WorkerThread.call(DataflowBatchWorkerHarness.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
```
**I can reliably reproduce the issue when I run it with IntelliJ Idea.** When I do `sbt runMain` everything works fine:

```bash
runMain com.reducted --runner=DataflowRunner --project=reducted --serviceAccount=reducted --region=us-west1
```

This makes me think it is somehow related to dependencies.

**Comment by jto** [comment]
Closing this issue since it's a duplicate of #2978. Please comment on #2978
